platform-sdk/docs/core/consensus-glossary.md

```diff
@@ -0,0 +1,199 @@
+# Consensus Glossary
+
+Note: this document is written assuming we are using birth rounds instead of generations to define
+when an event becomes ancient.
+
+## Ancient Event
+
+In plain words, an ancient event is an event that is old enough that we can stop caring about it. More formally,
+an event is considered to be ancient if its birth round is less than or equal to the
+[minimum non-ancient birth round](#minimum-non-ancient-birth-round).
+
+When computing consensus for a round, all ancient events are ignored.
+
+## Birth Round
+
+When a new event is created, it is assigned a birth round. A new event's birth round is equal to the
+[pending round](#pending-consensus-round) at the moment of its creation.
+
+### Invariant: Birth Rounds Never Decrease
+
+A child event is required to have a birth round that is not less than the birth round of any of its parents.
+
+### Claimed Parent Birth Rounds
+
+Each event must make a claim about the birth round of each of its parents, and that claim must be included in the
+hashed bytes of the event.
+
+If the event's claimed parents cause the event to violate the invariant that birth rounds never decrease, then the
+event is immediately discarded as invalid.
+
+The reason why the word "claimed" is used is because it's possible for a malicious node to lie about the existence
+of a parent that may not exist. The fact that a claimed parent birth round may not be correct is not a problem though.
+
+- if an attacker claims to have a parent that doesn't exist, we hold the child event in the orphan buffer until
+  the non-existent parent is ancient
+- if the attacker claims a parent has the wrong birth round and the parent exists, we detect the discrepancy and do not
```

Comment:
I think the case of an event claiming an incorrect birth round for a parent actually gets resolved before the linking phase, by the orphan buffer.
  
  Birth round is part of the `EventDescriptor`, so if `eventZ` says "my parent is `eventX`", but modifies the birth round of `eventX` -> `eventX_tampered`, then the `OrphanBuffer` will not be able to find the claimed parent, and `eventZ` will sit in the `OrphanBuffer` until `eventX_tampered` goes ancient

We do have logic in the `InOrderLinker` for if the birth round doesn't match. But I haven't been able to think of a way that such a mismatch could actually occur.